### PR TITLE
WPWebViewController Revamp!

### LINKED ISSUE #3722: WPWebViewController Cleanup
- [x] Nuke all of the (unused) properties
- [x] Refresh the User Interface to look a little bit more modern.

### PR DESCRIPTION
**Pull request by jleandroperez**
#### What's new:

- Bottom Toolbar has been nuked. We now have a UIRefreshControl that handles Pull to Refresh gestures.
- Back / Forward buttons are gone. If the user attempts to open an external link, the app will fall back to Safari (prior user's approval).
- UIActivitySpinner has been replaced by UIProgressView.
- Nuked a whole lot of code!.

@aerych + @diegoreymendez may i bother you with a review?. Although this was spoken via Slack, i'd like to triple confirm if the `Open in Safari` alertView is really what we want.

Thank you!

Closes #3722

--

#### Testing Scenario A: View Site
1. Launch WPiOS
2. Tap over **My Sites**
3. Pick any of your sites
4. Tap over the **View Site** row

As a result, WPWebViewController should be onscreen.

### CODE REVIEW

**Comment by diegoreymendez** [comment]
Reviewing now...

As a side not, following @sendhil's awesome post on [leaner PRs](https://coreappso2.wordpress.com/2015/05/13/lean-mean-pull-requests/) remember to try to provide testing instructions if there's a good testing path.

By sharing the testing path you're already using, reviewers can spend less time reviewing code (which is a good thing).

**Comment by jleandroperez** [comment]
@diegoreymendez right! sorry about that. I'll document (few) of the many ways to trigger a WebView.

Thanks!

File: WordPress/Classes/Utility/WPWebViewController.m [hunk 1]
```diff
@@ -9,19 +9,53 @@
...
-@class WPReaderDetailViewController;
+
+
+#pragma mark ====================================================================================
+#pragma mark Constants
+#pragma mark ====================================================================================
+
+#pragma mark ====================================================================================
+#pragma mark Constants
+#pragma mark ====================================================================================
```
Comment (diegoreymendez) [inline WordPress/Classes/Utility/WPWebViewController.m]: Double pragma header.

By the way, I really think we should try to standardize on pragmas like:

`#pragma mark - Constants`

Just a thought.

File: WordPress/Classes/Utility/WPWebViewController.m [hunk 2]
```diff
@@ -30,123 +64,43 @@ - (void)dealloc
...
-    if (self.url) {
-        [self refreshWebView];
-    } else {
-        self.navigationItem.title = NSLocalizedString(@"Loading...", @"");
-        [self.webView loadHTMLString:self.detailHTML baseURL:[NSURL URLWithString:@"https://en.wordpress.com"]];
-    }
-}
+    NSAssert(self.webView,          @"Missing Outlet!");
+    NSAssert(self.progressView,     @"Missing Outlet!");
+    NSAssert(self.optionsButton,    @"Missing Outlet!");
```
Comment (diegoreymendez) [inline WordPress/Classes/Utility/WPWebViewController.m]: Nice touch.

As a general rule when asserting properties I recommend asserting the ivars directly.  Even though right now the getters don't do any crazy stuff (like lazy initialization), they may in the future - thus possibly breaking here in a non-obvious way.

File: WordPress/Classes/Utility/WPWebViewController.m [hunk 3]
```diff
@@ -154,102 +108,39 @@ - (BOOL)expectsWidePanel
...
+    // Make sure we are not sharing URL like this: http://en.wordpress.com/reader/mobile/?v=post-16841252-1828
+    if ([permaLink rangeOfString:@"wordpress.com/reader/mobile/"].location != NSNotFound) {
+        permaLink = WPMobileReaderURL;
     }
 
     return permaLink;
 }
 
-- (NSString *)getDocumentTitle
+- (NSString *)documentTitle
```
Comment (diegoreymendez) [inline WordPress/Classes/Utility/WPWebViewController.m]: :+1:

**Comment by diegoreymendez** [comment]
@jleandroperez - I'm not seeing anything else at this moment.  Feature works nicely - as we discussed I'm also not too sure about the lack of a back button, but we should treat that separately imho.

Let me know when this is ready so I can approve.

File: WordPress/Classes/Utility/WPWebViewController.m (hunk 1 above)
Comment (jleandroperez) [inline WordPress/Classes/Utility/WPWebViewController.m]: You're right Diego, i'll change that. Thank you!

File: WordPress/Classes/Utility/WPWebViewController.m (hunk 2 above)
Comment (jleandroperez) [inline WordPress/Classes/Utility/WPWebViewController.m]: You're absolutely right. Thanks for this tip, never thought of it!

**Comment by jleandroperez** [comment]
@diegoreymendez comments addressed. Thanks for the awesome review Diego!

**Comment by diegoreymendez** [comment]
:shipit:

File: WordPress/Classes/Utility/WPWebViewController.m [hunk 4]
```diff
@@ -9,19 +9,43 @@
...
+static NSInteger const WPWebViewErrorAjaxCancelled          = -999;
+static NSInteger const WPWebViewErrorFrameLoadInterrupted   = 102;
+
+static CGFloat const WPWebViewProgressInitial               = 0.1f;
+static CGFloat const WPWebViewProgressFinal                 = 1.0f;
+
+static CGFloat const WPWebViewAnimationShortDuration        = 0.1f;
+static CGFloat const WPWebViewAnimationLongDuration         = 0.4f;
+static CGFloat const WPWebViewAnimationAlphaVisible         = 1.0f;
+static CGFloat const WPWebViewAnimationAlphaHidden          = 0.0f;
```
Comment (aerych) [inline WordPress/Classes/Utility/WPWebViewController.m]: We could probably leave off the `f` from the const values and let the compiler pick the correct type.

File: WordPress/Classes/Utility/WPWebViewController.m [hunk 5]
```diff
@@ -30,123 +54,43 @@ - (void)dealloc
...
-        self.backButton.width = (self.toolbar.frame.size.width / 2.0f) - 10.0f;
-        self.forwardButton.width = (self.toolbar.frame.size.width / 2.0f) - 10.0f;
-        UIBarButtonItem *spacer = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemFlexibleSpace target:nil action:nil];
-        NSArray *items = @[spacer, self.backButton, spacer, self.forwardButton, spacer];
-        self.toolbar.items = items;
-    }
-}
+    // TitleView
+    self.titleView                          = [NavigationTitleView new];
+    self.titleView.titleLabel.text          = NSLocalizedString(@"Loading...", @"Loading. Verb");
```
Comment (aerych) [inline WordPress/Classes/Utility/WPWebViewController.m]: Yay for adding localization comments :D

Comment (diegoreymendez) [inline WordPress/Classes/Utility/WPWebViewController.m]: When I read this I actually laughed out loud.

Comment (aerych) [inline WordPress/Classes/Utility/WPWebViewController.m]: :D

**Pull request merged**